src/form/form-input.tsx

```diff
@@ -25,9 +26,13 @@ const Component = (props: FormInputProps, ref: InputRef): JSX.Element => {
         ...otherProps
     } = props;
 
+    const [inputId] = useState(
+        () => id ?? `form-field-${SimpleIdGenerator.generate()}`
+    );
+
     return (
         <FormWrapper
-            id={id}
+            id={inputId}
```

Comment:
on further thought I think the main `data-testid` should be set on `FormWrapper`

this lets consumers query the main container, then they can further query the children as needed

there's also not much point to default as `id` because it could be dynamically generated now, which is useless for unit tests